modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsTest.ts

```diff
@@ -303,4 +304,24 @@ describe('browser.tinymce.core.textpatterns.TextPatternsTest', () => {
 
     editor.options.unset('text_patterns');
   });
+
+  it('TINY-9193: should move the caret to the next line after the Enter key pressed', () => {
+    const editor = hook.editor();
+    editor.options.set('text_patterns', [
+      { start: '~', end: '~', cmd: 'mceInsertContent', value: 'world' }
+    ]);
+    editor.setContent('<p>Hello</p><p>~test~</p>');
+    TinySelections.setCursor(editor, [ 1, 0 ], 6 );
```

Comment:
```suggestion
    editor.setContent('<p>Hello<br /></p><p>~test</p>');
    TinySelections.setCursor(editor, [ 1, 0 ], 5 );
    TinyContentActions.type(editor, '~');
```
I tried to run the test with the old code but it seems to not make the test fail.
With this close seems to fail without the fix and work with it but I'm not sure that is it the correct way to do it

Reply:
Did you try the test with the old code in Firefox/MacOS?

Reply:
actually in Firefox/Linux, but it seems to work as in Mac, now that you made me think, it is possible that in Linux it works differently